tests/Microsoft.Identity.Web.Test/CacheExtensionsTests.cs

```diff
@@ -116,12 +115,48 @@ public void DistributedCacheExtensions_NoCca_ThrowsException_Tests()
         [Fact]
         public void DistributedCacheExtensions_NoService_ThrowsException_Tests()
         {
-            CreateCca();
-            var ex = Assert.Throws<ArgumentNullException>(() => _confidentialApp.AddDistributedTokenCache(null!));
+            var confidentialApp = CreateCca();
+            var ex = Assert.Throws<ArgumentNullException>(() => confidentialApp.AddDistributedTokenCache(null!));
 
             Assert.Equal("initializeDistributedCache", ex.ParamName);
-        }
-
+        }
+
+        [Fact]
+        public async Task SingletonMsal_ResultsInCorrectCacheEntries_Test()
+        {
+            using MockHttpClientFactory mockHttpClient = new MockHttpClientFactory();
+            using (mockHttpClient.AddMockHandler(MockHttpCreator.CreateClientCredentialTokenHandler()))
+            using (mockHttpClient.AddMockHandler(MockHttpCreator.CreateClientCredentialTokenHandler()))
+            {
+                var confidentialApp = ConfidentialClientApplicationBuilder
+                               .Create(TestConstants.ClientId)
+                               .WithAuthority(TestConstants.AuthorityCommonTenant)
+                               .WithHttpClientFactory(mockHttpClient)
+                               .WithInstanceDiscovery(false)
+                               .WithClientSecret(TestConstants.ClientSecret)
+                               .Build();
+
+                var distributedCache = new TestDistributedCache();
+                confidentialApp.AddDistributedTokenCache(services =>
+                {
+                    services.AddSingleton<IDistributedCache>(distributedCache);
+                });
+
+                // Different tenants used to created different cache entries
+                var result1 = await confidentialApp.AcquireTokenForClient(new[] { TestConstants.s_scopeForApp })
+                    .WithTenantId("tenant1")

```

Comment:
>"tenant1" [](http://example.com/codeflow?start=34&length=9)

we have test constants you can use

Reply:
Put these into local constants, as they have to be in specific format.